### ISSUE TITLE
cuda.team_scan failing in nightly tests

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos#3607]
Following merge of PR #3536 many Cuda nightly builds began failing in the `cuda.team_scan` for the following tests:

Failing combos:
cuda/9.2 + Kepler35 arch (release)
cuda/9.2,cuda/10.1 + Pascal60,Power8 arch (release) 
cuda/10.1 + Volta70,Power9 arch (release)
cuda/10.1 + Volta70,SNB (release, debug)

Nightlies passed with the Cuda jobs having _both_ debugging and bounds checking enabled (enabling debugging alone was not sufficient).

**Sample output**:

cuda/10.1 Volta70,SNB
```bash
[ RUN      ] cuda.team_scan
Launching Test::TestTeamScan<Kokkos::Cuda, int> with M=0 and N=0...
Launching Test::TestTeamScan<Kokkos::Cuda, int> with M=0 and N=1...
Launching Test::TestTeamScan<Kokkos::Cuda, int> with M=1 and N=0...
:0: : block: [0,0,0], thread: [0,32,0] Assertion `Cuda::cuda_intra_block_scan requires power-of-two blockDim` failed.
...
```

**Sample reproducer instructions**:
```bash
module load cmake/3.12.3 cuda/10.1.243 gcc/7.4.0

$KOKKOS_PATH/generate_makefile.bash --with-devices=Cuda,Serial --arch=Power9,Volta70 --compiler=$KOKKOS_PATH/bin/nvcc_wrapper --cxxflags="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized " --cxxstandard="14" --with-cuda --kokkos-path=$KOKKOS_PATH --no-examples

# Corresponding cmake line:

cmake -DCMAKE_CXX_COMPILER=$KOKKOS_PATH/bin/nvcc_wrapper -DCMAKE_CXX_FLAGS="-g -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized" -DKokkos_ENABLE_OPENMP=ON -DKokkos_ENABLE_CUDA=ON -DKokkos_ARCH_VOLTA70=ON -DKokkos_ARCH_SNB=ON -DKokkos_ENABLE_TESTS=ON -DKokkos_ENABLE_EXAMPLES=OFF -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_CXX_EXTENSIONS=OFF -DCMAKE_CXX_STANDARD=14 -DCMAKE_BUILD_TYPE=RELEASE $KOKKOS_PATH

```

@jrmadsen @dalg24 @crtrott

**Comment by crtrott** [comment]
fixed in #3610

**Comment by ndellingwood** [comment]
The failing nightlies began passing last night after merge of #3610 :)
